src/uu/free/src/free.rs

```diff
@@ -280,6 +178,143 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     Ok(())
 }
 
+fn parse_print_format(matches: ArgMatches) -> impl Fn(&MemInfo) {
+    let wide = matches.get_flag("wide");
+    let human = matches.get_flag("human");
+    let si = matches.get_flag("si");
+    let total = matches.get_flag("total");
+    let lohi = matches.get_flag("lohi");
+    let committed = matches.get_flag("committed");
+    let one_line = matches.get_flag("line");
+
+    let convert = detect_unit(&matches);
+
+    // function that converts the number to the correct string
+    let n2s = move |x| {
+        if human {
+            humanized(x, si)
+        } else {
+            convert(x).to_string()
+        }
+    };
+
+    move |mem_info: &MemInfo| {
+        if one_line {
+            print_one_line(mem_info, &n2s);
+        } else {
+            if wide {
+                print_wide(mem_info, &n2s);
+            } else {
+                print(mem_info, &n2s);
+            }
+
+            if lohi {
+                print_lohi(mem_info, &n2s);
+            }
+
+            print_swap(mem_info, &n2s);
+
+            if total {
+                print_total(mem_info, &n2s);
+            }
+
+            if committed {
+                print_committed(mem_info, &n2s);
+            }
+        }
+    }
+}
+
+fn print_one_line(mem_info: &MemInfo, n2s: &dyn Fn(u64) -> String) {
```

Comment:
I think you can made this function just processing the content you wanna print, and print the result outside (such as `uu_main`?)